### ISSUE TITLE
go fails with "runtime error: index out of range" on i686

### ISSUE THREAD

**Issue by bcarlsson** [termux/termux-packages#217]
Error below is from a simple hello world program using only fmt.
But I am able to compile a go file with an empty main function.

It works on my other devices but not on my Xiaomi mipad 2 running on an intel z8500.

```
$ go build main.go
# command-line-arguments
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0xf77c4040, 0xd6824008)
        /data/data/com.termux/files/usr/lib/go/src/runtime/panic.go:464 +0x379
cmd/compile/internal/gc.mkvar(0xd6cca5d0, 0xd685a688, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/internal/gc/reg.go:435 +0x1590
cmd/compile/internal/gc.regopt(0xd685a1a0)
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/internal/gc/reg.go:1093 +0x3ccb
cmd/compile/internal/gc.compile(0xd6a38ae0)
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/internal/gc/pgen.go:521 +0xf46
cmd/compile/internal/gc.funccompile(0xd6a38ae0)
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/internal/gc/dcl.go:1450 +0x1ef
cmd/compile/internal/gc.Main()
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/internal/gc/lex.go:472 +0x23f4
cmd/compile/internal/x86.Main()
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/internal/x86/galign.go:108 +0x860
main.main()
        /data/data/com.termux/files/usr/lib/go/src/cmd/compile/main.go:30 +0x129
```

**Comment by fornwall** [comment]
Just a check - are you having the problem with the latest golang package (`go version` shows 1.6.1)?

I just tried building the following program on an intel device:

```go
package main
import "fmt"
func main() {
	fmt.Printf("Hello, world.\n")
}
```

This one worked for me. Is the above snippet enough to trigger your problem?

**Comment by bcarlsson** [comment]
Your snippet does compile and run. But if I change printf to println I get the error posted eatlier.

$ go version
go version go1.6.1 android/386

$ uname -a
Linux localhost 3.14.37-x86_64-266376-g97afc97 #1 SMP PREEMPT Sat Dec 19 00:06:19 CST 2015 x86_64 Android

**Comment by fornwall** [comment]
Thanks, I'm able to reproduce it with

```go
package main
import "fmt"
func main() {
	fmt.Println("Hello, world.\n")
}
```

**Comment by fornwall** [comment]
@bcarlsson Can you check if this is fixed in the latest go package for i686 (go 1.6.2)?
